common/dds-model.cpp

```diff
@@ -0,0 +1,332 @@
+
+        // Main Scrollable Section
+        ImGui::BeginChild("MainContent", ImVec2(w - 10, h - 120), true);
+        ImGui::PushItemWidth(150.0f);
+
+        // Connection Priority Section
+        priority connection_priority = classifyPriority(_changed_config.link.priority);
+        if (ImGui::CollapsingHeader("Connection Priority")) {
+            ImGui::Text("Select connection priority:");
+            ImGui::RadioButton("Ethernet First", reinterpret_cast<int*>(&connection_priority), 0);
+            if (static_cast<int>(connection_priority) == 0) {
+                ImGui::SameLine();
+                ImGui::SetCursorPosX(ImGui::GetCursorPosX() + 50);
+                ImGui::Text("Link Timeout (seconds)");
+                ImGui::SameLine();
+                int tempTimeout = static_cast<int>(_changed_config.link.timeout);
+                if (ImGui::InputInt("##Link Timeout (seconds)", &tempTimeout)) {
+                    _changed_config.link.timeout = static_cast<uint16_t>(std::max(0, tempTimeout));
+                }
+            }
+            ImGui::RadioButton("USB First", reinterpret_cast<int*>(&connection_priority), 1);
+            ImGui::RadioButton("Dynamic Priority", reinterpret_cast<int*>(&connection_priority), 2);
+            switch (connection_priority) {
+            case ETH_FIRST:
+                _changed_config.link.priority = link_priority::eth_first;
+                break;
+            case USB_FIRST:
+                _changed_config.link.priority = link_priority::usb_first;
+                break;
+            case DYNAMIC:
+                _changed_config.link.priority = _current_config.link.speed ? link_priority::dynamic_eth_first : link_priority::dynamic_usb_first;
+                break;
+            }
+        }
+
+        // Network Configuration Section
+        if (ImGui::CollapsingHeader("Network Configuration")) {
+            ImGui::Checkbox("Enable DHCP", &_changed_config.dhcp.on);
+            if (!_changed_config.dhcp.on) {
+                ImGui::Text("Static IP Address");
+                ImGui::SameLine();
+                float textbox_align = ImGui::GetCursorPosX();
+                ipInputText("Static IP Address", _changed_config.configured.ip);
+                ImGui::Text("Subnet Mask");
+                ImGui::SameLine();
+                ImGui::SetCursorPosX(textbox_align);
+                bool maskStylePushed = false;
+                ipInputText("Subnet Mask", _changed_config.configured.netmask);
+                ImGui::Text("Gateway");
+                ImGui::SameLine();
+                ImGui::SetCursorPosX(textbox_align);
+                ipInputText("Gateway", _changed_config.configured.gateway);
+            }
+            ImGui::Text("DHCP Timeout (seconds)");
+            ImGui::SameLine();
+            int tempTimeout = static_cast<int>(_changed_config.dhcp.timeout);
+            if (ImGui::InputInt("##DHCP Timeout (seconds)", &tempTimeout)) {
```

Comment:
Please place { on a new line for all uses here.
I will send you a mail about using clang format